### ISSUE TITLE
CosyVoice2 CosyVoice only inference_sft success  other inference as inference_zero_shot failed

### ISSUE THREAD

**Issue by HUSTHY** [FunAudioLLM/CosyVoice#1422]
**Describe the bug**
I pull the project from github by git clone https://github.com/FunAudioLLM/CosyVoice.git in 20250623

environment cuda:12.8   python 3.11   2.7.1+cu128

then inference_zero_shot inference like the readme wirte

cosyvoice = CosyVoice2("./CosyVoice2-0.5B") or cosyvoice = CosyVoice("./CosyVoice-300M-SFT")
cosyvoice.inference_zero_shot(tts_text = "收到好友从远方寄来的生日礼物，那份意外的惊喜与深深的祝福让我心中充满了甜蜜的快乐，笑容如花儿般绽放。", prompt_text='希望你以后能够做的比我还好呦。', prompt_speech_16k=self.prompt_speech_16k, zero_shot_spk_id='')
then  error occur in   /cosyvoice/flow/flow.py  line 254
token = self.input_embedding(torch.clamp(token, min=0)) * mask

error: 'indices' to have one of the following scalar types: Long, Int; but got torch.cuda.FloatTensor instead

then i correct it by add a code in /cosyvoice/flow/flow.py  line 254

toekn = token.long()
token = self.input_embedding(torch.clamp(token, min=0)) * mask

however another bug occur in /cosyvoice/cli/model.py  line 330
tts_speech, tts_source = self.hift.inference(speech_feat=tts_mel, cache_source=hift_cache_source)  
Calculated padded input size per channel: (2). Kernel size: (3). Kernel size can't be greater than actual input size

cosyvoice.inference_sft(input, '中文女') like this  success

I doubt that is other inference methods are updating and not finished, or inference_zero_shot  the way i use is wrong?

**Comment by aluminumbox** [comment]
follow code in readme, example in readme is verified

**Comment by HUSTHY** [comment]
已收到！谢谢！                          ——黄洋

**Comment by HUSTHY** [comment]
the code i supplied follow readme code strictly, but it does not work. 
however i change the code 
def frontend_zero_shot(self, tts_text, prompt_text, prompt_speech_16k, resample_rate, zero_shot_spk_id):
        tts_text_token, tts_text_token_len = self._extract_text_token(tts_text)
        if zero_shot_spk_id == '':
            prompt_text_token, prompt_text_token_len = self._extract_text_token(prompt_text)
            prompt_speech_resample = torchaudio.transforms.Resample(orig_freq=16000, new_freq=resample_rate)(prompt_speech_16k)
            speech_feat, speech_feat_len = self._extract_speech_feat(prompt_speech_resample)
            speech_token, speech_token_len = self._extract_speech_token(prompt_speech_16k)

            if resample_rate == 24000:
                # cosyvoice2, force speech_feat % speech_token = 2
                token_len = min(int(speech_feat.shape[1] / 2), speech_token.shape[1])
                speech_feat, speech_feat_len[:] = speech_feat[:, :2 * token_len], 2 * token_len
                speech_token, speech_token_len[:] = speech_token[:, :token_len], token_len
            embedding = self._extract_spk_embedding(prompt_speech_16k)
            model_input = {'prompt_text': prompt_text_token, 'prompt_text_len': prompt_text_token_len,
                           'llm_prompt_speech_token': speech_token, 'llm_prompt_speech_token_len': speech_token_len,
                           'flow_prompt_speech_token': speech_token, 'flow_prompt_speech_token_len': speech_token_len,
                           'prompt_speech_feat': speech_feat, 'prompt_speech_feat_len': speech_feat_len,
                           'llm_embedding': embedding, 'flow_embedding': embedding}
        else:
            model_input = self.spk2info[zero_shot_spk_id]
            model_input['text'] = tts_text_token
            model_input['text_len'] = tts_text_token_len
        return model_input

to

def frontend_zero_shot(self, tts_text, prompt_text, prompt_speech_16k, resample_rate, zero_shot_spk_id):
        tts_text_token, tts_text_token_len = self._extract_text_token(tts_text)
        if zero_shot_spk_id == '':
            prompt_text_token, prompt_text_token_len = self._extract_text_token(prompt_text)
            prompt_speech_resample = torchaudio.transforms.Resample(orig_freq=16000, new_freq=resample_rate)(prompt_speech_16k)
            speech_feat, speech_feat_len = self._extract_speech_feat(prompt_speech_resample)
            speech_token, speech_token_len = self._extract_speech_token(prompt_speech_16k)

            print(f"speech_token {speech_token}")
            print(f"speech_token {speech_token.dtype}")
            if resample_rate == 24000:
                # cosyvoice2, force speech_feat % speech_token = 2
                token_len = min(int(speech_feat.shape[1] / 2), speech_token.shape[1])
                speech_feat, speech_feat_len[:] = speech_feat[:, :2 * token_len], 2 * token_len
                speech_token, speech_token_len[:] = speech_token[:, :token_len], token_len
            embedding = self._extract_spk_embedding(prompt_speech_16k)
            model_input = {'prompt_text': prompt_text_token, 'prompt_text_len': prompt_text_token_len,
                           'llm_prompt_speech_token': speech_token, 'llm_prompt_speech_token_len': speech_token_len,
                           'flow_prompt_speech_token': speech_token, 'flow_prompt_speech_token_len': speech_token_len,
                           'prompt_speech_feat': speech_feat, 'prompt_speech_feat_len': speech_feat_len,
                           'llm_embedding': embedding, 'flow_embedding': embedding}
            
            model_input['text'] = tts_text_token //增加这个行
            model_input['text_len'] = tts_text_token_len  //增加这个行
        else:
            model_input = self.spk2info[zero_shot_spk_id]
            model_input['text'] = tts_text_token
            model_input['text_len'] = tts_text_token_len
        print(f"model_input {model_input}")
        return model_input

it works good!

**Comment by HUSTHY** [comment]
已收到！谢谢！                          ——黄洋
